src/PolyType.Roslyn/Model/AssociatedTypeModel.cs

```diff
@@ -8,4 +8,33 @@ namespace PolyType.Roslyn;
 /// <param name="AssociatedType">The symbol for the related type.</param>
 /// <param name="AssociatingAssembly">The assembly that declared the association.</param>
 /// <param name="Location">The location where the association is declared.</param>
-public record struct AssociatedTypeModel(INamedTypeSymbol AssociatedType, IAssemblySymbol AssociatingAssembly, Location? Location);
+/// <param name="Requirements">The associated type's requirements.</param>
+public record struct AssociatedTypeModel(INamedTypeSymbol AssociatedType, IAssemblySymbol AssociatingAssembly, Location? Location, TypeShapeDepth Requirements);
+
+/// <summary>
+/// Describes the requirements for preparing an associated type.
+/// </summary>
+/// <devremarks>
+/// Keep this in sync with the TypeShapeDepth enum defined in the PolyType assembly.
+/// </devremarks>
+[Flags]
+public enum TypeShapeDepth
+{
+    /// <summary>No shape is required.</summary>
+    None = 0x0,
+
+    /// <summary>
+    /// A constructor should be included in the shape, if one is declared.
```

Comment:
Would this also include parameterized constructors and its parameter types?

Reply:
I presume from the definition of `All` that the above is true, therefore:
```suggestion
    /// A constructor , its parameters and their types should be included in the shape, if one is declared.
```